clang/test/ParserHLSL/hlsl_resource_class_attr.hlsl

```diff
@@ -0,0 +1,38 @@
+// RUN: %clang_cc1 -triple dxil-pc-shadermodel6.0-compute -x hlsl -ast-dump -o - %s | FileCheck %s
+
+
+// CHECK: -HLSLResourceClassAttr 0x{{[0-9a-f]+}} <<invalid sloc>> SRV
+struct [[hlsl::resource_class(SRV)]] Eg1 {

```

Comment:
I thought that the idea was that the attribute would be applied to the handle type - ie to a _member_ of a struct, not to the struct itself?

Reply:
For the purpose of these tests it shouldn't matter very much - we can just apply the attribute to whatever since we're just checking that the AST picks up the attribute correctly.

Also as far as things like RWBuffer go this change doesn't actually move the attribute to the handle member, that's a later part of #98192

Reply:
Ah, ok, I see this is step 1 of a longer journey.  The PR description says the attribute can only be applied to structs - I guess this is something that'll need to be changed as well.

Reply:
Ah, I hadn't noticed the "struct" subject on the attr definition. It's probably ok for now, but it'll definitely need to change.